src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,325 @@
+    mapping(bytes32 => QuorumBitmapUpdate[]) internal _operatorIdToQuorumBitmapHistory;
+    /// @notice the mapping from operator's address to the operator struct
+    mapping(address => Operator) internal _operators;
+    /// @notice the dynamic-length array of the registries this coordinator is coordinating
+    address[] public registries;
+
+    modifier onlyServiceManagerOwner {
+        require(msg.sender == serviceManager.owner(), "BLSIndexRegistryCoordinator.onlyServiceManagerOwner: caller is not the service manager owner");
+        _;
+    }
+
+    constructor(
+        ISlasher _slasher,
+        IServiceManager _serviceManager,
+        IStakeRegistry _stakeRegistry,
+        IBLSPubkeyRegistry _blsPubkeyRegistry,
+        IIndexRegistry _indexRegistry
+    ) {
+        slasher = _slasher;
+        serviceManager = _serviceManager;
+        stakeRegistry = _stakeRegistry;
+        blsPubkeyRegistry = _blsPubkeyRegistry;
+        indexRegistry = _indexRegistry;
+    }
+
+    function initialize(OperatorSetParam[] memory _operatorSetParams) external initializer {
+        // the stake registry is this contract itself
+        registries.push(address(stakeRegistry));
+        registries.push(address(blsPubkeyRegistry));
+        registries.push(address(indexRegistry));
+
+        // set the operator set params
+        require(IVoteWeigher(address(stakeRegistry)).quorumCount() == _operatorSetParams.length, "BLSIndexRegistryCoordinator: operator set params length mismatch");
+        for (uint8 i = 0; i < _operatorSetParams.length; i++) {
+            quorumOperatorSetParams[i] = _operatorSetParams[i];
+        }
+    }
+
+    /// @notice Returns task number from when `operator` has been registered.
+    function getFromTaskNumberForOperator(address operator) external view returns (uint32) {
+        return _operators[operator].fromTaskNumber;
+    }
+
+    /// @notice Returns the operator struct for the given `operator`
+    function getOperator(address operator) external view returns (Operator memory) {
+        return _operators[operator];
+    }
+
+    /// @notice Returns the operatorId for the given `operator`
+    function getOperatorId(address operator) external view returns (bytes32) {
+        return _operators[operator].operatorId;
+    }
+
+    /// @notice Returns the quorum bitmap for the given `operatorId` at the given `blockNumber` via the `index`
+    function getQuorumBitmapByOperatorIdAtBlockNumberByIndex(bytes32 operatorId, uint32 blockNumber, uint256 index) external view returns (uint192) {
+        QuorumBitmapUpdate memory quorumBitmapUpdate = _operatorIdToQuorumBitmapHistory[operatorId][index];
+        require(
+            quorumBitmapUpdate.updateBlockNumber <= blockNumber, 
+            "BLSRegistryCoordinator.getQuorumBitmapByOperatorIdAtBlockNumberByIndex: quorumBitmapUpdate is from after blockNumber"
+        );
+        require(
+            quorumBitmapUpdate.nextUpdateBlockNumber > blockNumber, 
```

Comment:
are we initializing `nextUpdateBlockNumber` as the max uint32? otherwise this will revert for the latest entry, right?
The fact I'm asking this here means we should presumably improve the documentation throughout (not just here).

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/ba096da06717b4fed400ed1348b2c3a0c677cc59
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/b20e598e84e6619f04036a6adeb616be4aad2d87